src/StreamJsonRpc.Shared/Reflection/MethodNameAttribute.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using Microsoft;
+
+namespace StreamJsonRpc
+{
+    /// <summary>
+    /// Attribute which allows a method to have an alternate name.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = false)]
```

Comment:
I tested the behavior of this, even if I set AllowMultiple to false the overridden method can still specify its own MethodName attribute with a different name value.  Not sure what the behavior should be for setting AllowMultiple.  Hence the extra check and unit tests.